pkg/helpers/sa_syncer.go

```diff
@@ -0,0 +1,99 @@
+package helpers
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	coreclientv1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+)
+
+// EnsureSAToken get the saToken of target sa and ensure it was rendered as expected.
+// A usage example combined with RenderToKubeconfigSecert would be like the following:
+// ...
+// saName := clusterManagerName+"-registration-controller-sa"
+// secretName := saName+"-kubeconfig"
+// err := helpers.EnsureSAToken(ctx, saName, clustermanagerNamespace, amdinKubeClient,
+// 	helpers.RenderToKubeconfigSecret(secretName, clustermanagerNamespace, AdminKubeconfig, kubeClient, recorder)) // kubeClient used to create secret; normally recorder is from controller object.
+// if err != nil {
+// 	return err
+// }
+// ...
+func EnsureSAToken(ctx context.Context, saName, saNamespace string, client kubernetes.Interface, renderSAToken func([]byte) error) error {
+	// get the service account
+	sa, err := client.CoreV1().ServiceAccounts(saNamespace).Get(ctx, saName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	if len(sa.Secrets) == 0 {
+		return fmt.Errorf("token secret for %s not exist yet", saName)
+	}
+	for _, secret := range sa.Secrets {
+		if strings.Contains(secret.Name, "token") {
```

Comment:
I check that the 0 index of secret in sa is not always saName-token-xxx, it could be saName-dockercfg-xx.
what if there is a "token" string in the sa name?
I think you can ref this https://github.com/open-cluster-management/managedcluster-import-controller/blob/884905557ba9e956ec7f975c01c72a7e6d577307/pkg/controller/importconfig/cluster_info.go#L37

Reply:
Thanks for the feedback, it's a great reference. Now we set triple checks to make sure we get the correct result.
1. prefix check
2. token in name check
3. secret type check